### PR TITLE
feat: Adjust repository to create single table for pages

### LINKED ISSUE #131: feat: Use single page table

### PR DESCRIPTION
**Pull request by dricazenck**
As part of POC using SurrealDB, we decided to create single table instead one for each page. 
 
The pages should have single table with content as string
The content will be json string and the service should take care of serialization and de-serialization

Also created Delete endpoint and Get all pages

image

resolve #131

### CODE REVIEW

File: src/main/java/com/wcc/platform/service/PlatformService.java [hunk 1]
```diff
@@ -42,12 +46,24 @@ public ResourceContent saveResourceContent(final ResourceContent resourceContent
...
+    } catch (JsonProcessingException e) {
+      throw new PlatformInternalException(PageType.LANDING_PAGE, e);
+    }
   }
 
   /** Save any type of page based on page Type. */
   public Object savePage(final FooterPage page) {
-    return footerRepository.save(page);
+    try {
+      return pageRepository.save(objectMapper.writeValueAsString(page));
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/service/PlatformService.java:59]: TODO: Investigate why it is not saving =/

Comment (nverbic) [reply src/main/java/com/wcc/platform/service/PlatformService.java:59]: Does the saving work for the landing page above?

File: src/test/java/com/wcc/platform/service/CmsServiceTest.java [hunk 2]
```diff
@@ -33,7 +32,7 @@ class CmsServiceTest {
   void setUp() {
     MockitoAnnotations.openMocks(this);
 
-    service = new CmsService(objectMapper, footerRepository, landingPageRepository);
+    service = new CmsService(objectMapper, pageRepository);
```
Comment (goelsonali) [inline src/test/java/com/wcc/platform/service/CmsServiceTest.java:35]: Do we need to add a test to use mock pageRepository to fetch the data from repository + if not found then fallback ?

Comment (dricazenck) [reply src/test/java/com/wcc/platform/service/CmsServiceTest.java:35]: It is missing the test for the success case using the surrealDB, now we are only testing the fallback behaviour. I will do that in another issue: [#128](https://github.com/Women-Coding-Community/wcc-backend/issues/128)

**Pull request merged**